### PR TITLE
[Feature] Implemented chat initialization in ChatService

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Initialize ChatClient and ChatThreadClient in through redux to ChatService and unit test
* Implemented skeleton for Repository state/action/reducer
* Added common code for Test mocking and unit test

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift [hunk 1]
```diff
@@ -29,9 +29,13 @@ class ChatSDKWrapper: NSObject, ChatSDKWrapperProtocol {
         logger.debug("CallingSDKWrapper deallocated")
     }
 
-    func initializeChat() async throws {
+    func initializeChat() async throws -> String {
```
Comment (palatter) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift:32]: Seems unclear from the naming of this, that the topic name would be returned.

Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift:32]: In order to check if it's a valid token need to make a request to SDK, so might as well try to get thread property (the topic). Is there a suggested naming that you think will indicate both initialization and returning the topic?
`initializeChatWithTopic` sounds more like providing a topic for initialization, and `initializeChatAndGetChatTopic` sounds a bit odd.

Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift:32]: Spoke with Inderpal, will make this function void, and have side effect to send the topic through Subjects, will update the code

**Review by jpeng-ms — APPROVED** [review]
looks good

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift (hunk 1 above)
Comment (palatter) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Service/Chat/ChatSDKWrapper.swift]: Sounds good!

**Comment by palatter** [comment]
Looks good!

**Review by palatter — APPROVED** [review]

**Pull request merged**